model/src/main/java/org/eclipse/rdf4j/model/impl/SimpleIRI.java

```diff
@@ -115,7 +115,20 @@ public boolean equals(Object o) {
 		}
 
 		if (o instanceof IRI) {
-			return toString().equals(o.toString());
+
+			String a = toString();
+			String b = o.toString();
+
+			if (a.length() != b.length()) {
+				return false;
+			}
+
+			// faster if using same namespace and UUID
+			if (a.length() > 0 && a.charAt(a.length() - 1) != b.charAt(b.length() - 1)) {
```

Comment:
I don't have any benchmarks for this, so it's more an assumption that anything else.